main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"time"
+
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+
+	"github.com/ONSdigital/dp-frontend-articles-controller/config"
+	"github.com/ONSdigital/dp-frontend-articles-controller/routes"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
+)
+
+func main() {
+	log.Namespace = "dp-frontend-articles-controller"
+	cfg, err := config.Get()
+	ctx := context.Background()
+	if err != nil {
+		log.Event(ctx, "unable to retrieve service configuration", log.Error(err))
+		os.Exit(1)
+	}
+
+	log.Event(ctx, "got service configuration", log.Data{"config": cfg})
+
+	versionInfo, err := health.NewVersionInfo(
+		BuildTime,
+		GitCommit,
+		Version,
+	)
+
+	r := mux.NewRouter()
```

Comment:
Interesting - the project generation tool needs to create a service layer for component tests. Basically a lot of this logic should exist in a service package.

I wouldn't do anything for this in this pull request but it might be worth bringing in a new story to rework main.go and create that service package. Possibly update project generation tool templates too (20% time ticket)